### PR TITLE
chore: extract downloadAndSha to a separate class to reused it

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
for https://github.com/containers/podman-desktop/issues/6360 I need to validate some github artifacts, and then having a dedicated class will help

### What issues does this PR fix or reference?
related to for https://github.com/containers/podman-desktop/issues/6360 

### How to test this PR?

build should be 🟢 
- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: extensions/podman/scripts/podman-download.spec.ts [hunk 1]
```diff
@@ -63,6 +66,10 @@ class TestPodmanDownload extends PodmanDownload {
   public getShaCheck(): ShaCheck {
     return super.getShaCheck();
   }
+
+  public getDownloadAndCheck(): DownloadAndCheck {
+    return super.getDownloadAndCheck();
```
Comment (dgolovin) [inline extensions/podman/scripts/podman-download.spec.ts:71]: Just curious, why make method from base class protected, then override it in ancestor and make it public? What is the idea here?

**Review by dgolovin — APPROVED** [review]
LGTM

File: extensions/podman/scripts/podman-download.spec.ts (hunk 1 above)
Comment (benoitf) [reply extensions/podman/scripts/podman-download.spec.ts:71]: to call the method directly from a unit test (but not being able to call it directly when we instantiate the class in the code base)

Comment (benoitf) [reply extensions/podman/scripts/podman-download.spec.ts:71]: (or to replace the method during test)

**Pull request merged**